weather_mv/loader_pipeline/bq.py

```diff
@@ -309,3 +330,49 @@ def fetch_geo_point(lat: float, long: float) -> str:
     long = ((long + 180) % 360) - 180
     point = geojson.dumps(geojson.Point((long, lat)))
     return point
+
+
+def fetch_geo_polygon(latitude: float, longitude: float, lat_grid_resolution: float, lon_grid_resolution: float) -> str:
+    """Create a Polygon based on latitude, longitude and resolution."""
+    if longitude >= 180:
+        longitude = longitude - 360
+    lat_lon_bound = bound_point(latitude, longitude, lat_grid_resolution, lon_grid_resolution)
+    polygon = geojson.dumps(geojson.Polygon([
+        (lat_lon_bound[0][0], lat_lon_bound[0][1]),  # lower_left
+        (lat_lon_bound[1][0], lat_lon_bound[1][1]),  # upper_left
+        (lat_lon_bound[2][0], lat_lon_bound[2][1]),  # upper_right
+        (lat_lon_bound[3][0], lat_lon_bound[3][1]),  # lower_right
+        (lat_lon_bound[0][0], lat_lon_bound[0][1]),  # lower_left
+    ]))
+    return polygon
+
+
+def bound_point(latitude, longitude, lat_grid_resolution, lon_grid_resolution) -> t.List:
+    '''Calculate the bound point based on latitude, longitude and grid resolution.'''
+    is_lat_out_of_bound = True if latitude in [90.0, -90.0] else False
+    is_lon_out_of_bound = True if longitude in [-180.0, 180.0] else False
+
+    lat_range = get_lat_lon_range(latitude, "latitude", is_lat_out_of_bound,
+                                  lat_grid_resolution, lon_grid_resolution)
+    lon_range = get_lat_lon_range(longitude, "longitude", is_lon_out_of_bound,
+                                  lat_grid_resolution, lon_grid_resolution)
+    lower_left = [lon_range[1], lat_range[1]]
+    upper_left = [lon_range[0], lat_range[1]]
+    upper_right = [lon_range[0], lat_range[0]]
+    lower_right = [lon_range[1], lat_range[0]]
+    return [lower_left, upper_left, upper_right, lower_right]
+
+
+def get_lat_lon_range(value, lat_lon, is_point_out_of_bound, lat_grid_resolution, lon_grid_resolution):
```

Comment:
I think the logic for normalizing the lat/lng ranges to spread out across too many places. Can we correct for it once near the logical beginning? 

For example, can we just compute the following line before getting the input lat/lng values? 

https://github.com/google/weather-tools/pull/337/files#diff-938d4c3999aa9d0bd9f832240de92dc8ea660bf71f8b9b6d751b3a16773ce4b1L309

Reply:
Code updated as per your suggestion. Thanks for the input!